### PR TITLE
scPipe: modifications

### PR DESCRIPTION
**Pull request by mblue9**
Replaces https://github.com/galaxyproject/tools-iuc/pull/2046

@bgruening I've added BAM input for your user

FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

File: tools/scpipe/scpipe.xml [hunk 1]
```diff
@@ -24,153 +26,207 @@ echo $(R --version | grep version | grep -v GNU)", scPipe version" $(R --vanilla
 
 ## Link input files
 
-#if $ref_fasta.fasta_source == "history":
-    #set $fasta_name = re.sub('[^\w\-\s]', '_', str($ref_fasta.ref_fa_hist.element_identifier))
-    ln -s '$ref_fasta.ref_fa_hist' '$fasta_name' &&
+#if $samples.format_select == "bam":
+    #set $bam_name = re.sub('[^\w\-\s]', '_', str($samples.bam.element_identifier))
+    ln -s '$samples.bam' '$bam_name' &&
```
Comment (bgruening) [inline tools/scpipe/scpipe.xml]: You probably want to symlink the indices as well?
```
ln -s '$samples.bam' '${bam_name}.bam'
ln -s '$samples.bam.metadata.bam_index' '${bam_name}.bai'
```

Comment (mblue9) [reply tools/scpipe/scpipe.xml]: Ok, just wondering how come it works currently without that?

**Comment by mblue9** [comment]
Ok I can see if I can make the test file smaller, just didn't know if it was worth the time.

File: tools/scpipe/scpipe.xml (hunk 1 above)
Comment (bgruening) [reply tools/scpipe/scpipe.xml]: Maybe it does not use the index? Or maybe it creates its own on the fly? I do not know.

**Comment by bgruening** [comment]
Waiting for the new release or merging? @mblue9 can I squash the commits on merge or do you want to sqash to get rid of the big BAM file?

**Comment by mblue9** [comment]
When is the new release? If your user is looking for it I'd just go ahead and merge it and that's fine you can squash.

**Comment by bgruening** [comment]
I hope we will switch the labels soon on bioconda, but I can not say for sure.

**Pull request merged**